awswrangler/elasticsearch/_write.py

```diff
@@ -0,0 +1,285 @@
+"""Amazon Elasticsearch Write Module (PRIVATE)."""
+
+import json
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Mapping, Optional, Union, Tuple, Iterable
+
+import boto3
+import pandas as pd
+
+from elasticsearch import Elasticsearch
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def create_index(
+    index: str,
+    doc_type: Optional[str] = None,
+    settings: Optional[Dict[str, Any]] = None,
+    mappings: Optional[Dict[str, Any]] = None,
+    boto3_session: Optional[boto3.Session] = None,
+    con: Optional[Elasticsearch] = None
+) -> Dict[str, Any]:
+    """Creates an index.
+
+    Parameters
+    ----------
+    index : str
+        Name of the index.
+    doc_type : str
+        Name of the document type (for Elasticsearch versions 5.x and earlier).
+    settings : Dict[str, Any], optional
+        Index settings
+        https://opensearch.org/docs/opensearch/rest-api/create-index/#index-settings
+    mappings : Dict[str, Any], optional
+        Index mappings
+        https://opensearch.org/docs/opensearch/rest-api/create-index/#mappings
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 Session will be used if boto3_session receive None.
+    con : elasticsearch.Elasticsearch, optional
+        Elasticsearch client. A new connection will be established if con receive None.
+
+    Returns
+    -------
+    Dict[str, Any]
+        Elasticsearch rest api response
+        https://opensearch.org/docs/opensearch/rest-api/create-index/#response.
+
+    Examples
+    --------
+    Creating an index.
+
+    >>> import awswrangler as wr
+    >>> response = wr.elasticsearch.create_index(
+    ...     index="sample-index1",
+    ...     mappings={
+    ...        "properties": {
+    ...          "age":  { "type" : "integer" }
+    ...        }
+    ...     },
+    ...     settings={
+    ...         "index": {
+    ...             "number_of_shards": 2,
+    ...             "number_of_replicas": 1
+    ...          }
+    ...     }
+    ... )
+
+    """
+
+
+def index_json(
+    path: Union[str, Path],
+    index: str,
+    doc_type: Optional[str] = None,
+    bulk_params: Optional[Union[List[Any], Tuple[Any], Dict[Any, Any]]] = None,
+    boto3_session: Optional[boto3.Session] = None,
+    **kwargs
+) -> Dict[str, Any]:
+    """Index all documents from JSON file to Elasticsearch index.
+
+    The JSON file should be in a JSON-Lines text format (newline-delimited JSON) - https://jsonlines.org/.
+
+    Parameters
+    ----------
+    path : Union[str, Path]
+        Path as str or Path object to the JSON file which contains the documents.
```

Comment:
Just confirming that this would be an S3 path?

Reply:
I'm thinking it can be either local or s3 path. what do you think?
I've copied the description from [dynamodb.put_json](https://aws-data-wrangler.readthedocs.io/en/2.10.0/stubs/awswrangler.dynamodb.put_json.html#awswrangler.dynamodb.put_json)
But I agree it should be specific

Reply:
Fair enough, we should probably do both 👍🏼